apps/arena/lib/arena/bots/bot.ex

```diff
@@ -0,0 +1,146 @@
+defmodule Arena.Bots.Bot do
+  @moduledoc """
+  Arena bot GenServer.
+  This module is in charge of handling bots messages to the Game process.
+  """
+  use GenServer
+  alias Phoenix.PubSub
+  alias Arena.Bots.PathfindingGrid
+  alias BotManager.BotStateMachine
+  alias BotManager.BotStateMachineChecker
+  require Logger
+  @action_delay_ms 30
+
+  def start_link(%{bot_id: bot_id, game_id: _game_id} = params) do
+    GenServer.start_link(__MODULE__, params, name: generate_bot_name(bot_id))
```

Comment:
Why are we using `Registry`? I noticed we're only using it here, but we never reference the bot by its name

Reply:
Since each bot is a different GenServer instance, I just wanted to name them uniquely: https://hexdocs.pm/elixir/1.12/GenServer.html#module-name-registration